driver-core/src/main/java/com/datastax/driver/core/CodecRegistry.java

```diff
@@ -476,9 +488,14 @@ public CodecRegistry register(Iterable<? extends TypeCodec<?>> codecs) {
         checkNotNull(cqlType, "Parameter cqlType cannot be null");
         if (logger.isTraceEnabled())
             logger.trace("Querying cache for codec [{} <-> {}]", toString(cqlType), toString(javaType));
-        CacheKey cacheKey = new CacheKey(cqlType, javaType);
         try {
-            TypeCodec<?> codec = cache.get(cacheKey);
+            TypeCodec<?> codec = PRIMITIVE_CODECS.get(cqlType);
+            if (codec != null && (javaType == null || codec.accepts(javaType))) {
+                logger.trace("Returning cached codec {}", codec);
```

Comment:
Trivial: we could log "Returning primitive codec {}" here, to distinguish from the other case.